# Titanic Dataset – Task 1: Understanding Dataset & Data Types


## 1️⃣ Dataset Loading & Structure

- Dataset contains **891 rows and 12 columns**
- Each row represents a **passenger**
- Loaded using `pandas.read_csv()`
- First and last few records were inspected to understand structure

### Key Columns
PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, Embarked


In [1]:

import pandas as pd
df = pd.read_csv("Titanic-Dataset.csv")
df.head(), df.tail()


(   PassengerId  Survived  Pclass  \
 0            1         0       3   
 1            2         1       1   
 2            3         1       3   
 3            4         1       1   
 4            5         0       3   
 
                                                 Name     Sex   Age  SibSp  \
 0                            Braund, Mr. Owen Harris    male  22.0      1   
 1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
 2                             Heikkinen, Miss. Laina  female  26.0      0   
 3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
 4                           Allen, Mr. William Henry    male  35.0      0   
 
    Parch            Ticket     Fare Cabin Embarked  
 0      0         A/5 21171   7.2500   NaN        S  
 1      0          PC 17599  71.2833   C85        C  
 2      0  STON/O2. 3101282   7.9250   NaN        S  
 3      0            113803  53.1000  C123        S  
 4      0            373450   8.0500


## 2️⃣ Data Types Identification

### 🔢 Numerical Features
- PassengerId  
- Age  
- Fare  
- SibSp  
- Parch  

### 🏷 Categorical Features
- Sex  
- Ticket  
- Cabin  
- Embarked  
- Name  

### 🔘 Binary Feature
- Survived (0 = Not Survived, 1 = Survived)

### 📊 Ordinal Feature
- Pclass (1st, 2nd, 3rd class – ordered categories)



## 3️⃣ Data Info & Statistics


In [2]:

df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB



### df.info() Observations
- Age has missing values (**177 nulls**)
- Cabin has heavy missing values (**687 nulls**)
- Embarked has **2 missing values**
- Dataset size: **83.7 KB** (small but ideal for ML learning)


In [3]:

df.describe(include='all')


PassengerId    Survived      Pclass                     Name   Sex  \
count    891.000000  891.000000  891.000000                      891   891   
unique          NaN         NaN         NaN                      891     2   
top             NaN         NaN         NaN  Braund, Mr. Owen Harris  male   
freq            NaN         NaN         NaN                        1   577   
mean     446.000000    0.383838    2.308642                      NaN   NaN   
std      257.353842    0.486592    0.836071                      NaN   NaN   
min        1.000000    0.000000    1.000000                      NaN   NaN   
25%      223.500000    0.000000    2.000000                      NaN   NaN   
50%      446.000000    0.000000    3.000000                      NaN   NaN   
75%      668.500000    1.000000    3.000000                      NaN   NaN   
max      891.000000    1.000000    3.000000                      NaN   NaN   

               Age       SibSp       Parch  Ticket        Fare    Cabin  \
count   714.000000  891.000000  891.000000     891  891.000000      204   
unique         NaN         NaN         NaN     681         NaN      147   
top            NaN         NaN         NaN  347082         NaN  B96 B98   
freq           NaN         NaN         NaN       7         NaN        4   
mean     29.699118    0.523008    0.381594     NaN   32.204208      NaN   
std      14.526497    1.102743    0.806057     NaN   49.693429      NaN   
min       0.420000    0.000000    0.000000     NaN    0.000000      NaN   
25%      20.125000    0.000000    0.000000     NaN    7.910400      NaN   
50%      28.000000    0.000000    0.000000     NaN   14.454200      NaN   
75%      38.000000    1.000000    0.000000     NaN   31.000000      NaN   
max      80.000000    8.000000    6.000000     NaN  512.329200      NaN   

       Embarked  
count       889  
unique        3  
top           S  
freq        644  
mean        NaN  
std         NaN  
min         NaN  
25%         NaN  
50%         NaN  
75%         NaN  
max         NaN


### df.describe() Insights
- Mean age ≈ **29.7 years**
- Fare is highly **right-skewed** (max = 512)
- Survival rate ≈ **38.4%**



## 4️⃣ Categorical Distribution

- **Sex:** Male dominant (~65%)

### Embarked
- S (Southampton) – majority  
- C (Cherbourg)  
- Q (Queenstown)



## 5️⃣ Target Variable & Features

### 🎯 Target Variable
- **Survived**

### 📥 Input Features
Pclass, Sex, Age, SibSp, Parch, Fare, Embarked

These features are suitable for **classification models**.



## 6️⃣ Dataset Suitability for Machine Learning

### ✅ Suitable for ML because:
- Clear target variable
- Mix of numerical & categorical features
- Manageable size for experimentation
- Real-world data quality issues present

### ⚠ Requires preprocessing:
- Missing value handling (Age, Cabin)
- Encoding categorical variables
- Feature scaling (Fare)
